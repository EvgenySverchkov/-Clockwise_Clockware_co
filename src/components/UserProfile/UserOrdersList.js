import React from "react";
import { useDispatch, useSelector } from "react-redux";

import List from "../List";
import {initUserOrders, changeUserOrdersListIsLoad} from "../../store/ordersManagement/actions"
import { SERVERDOMAIN } from "../../services/serverUrls";

const UserOrdersList = ({ history }) => {
  const dispatch = useDispatch();
  const state = useSelector((state) => {
    return {
      userOrders: state.ordersReducer.userOrders,
      userOrdersListIsLoad: state.ordersReducer.userOrdersListIsLoad,
    };
  });
  function getUserOrdersFromServerToState() {
    const headers = {
      Authorization: "Bearer " + localStorage.getItem("token") || "",
      "Content-Type": "application/json",
    };
    dispatch(changeUserOrdersListIsLoad(true));
    fetch(`${SERVERDOMAIN}/orders/getUserOrders`, {
      headers,
      method: "POST",
      body: localStorage.getItem("user"),
    })
      .then((json) => json.json())
      .then((data) => {
        dispatch(initUserOrders(data));
        dispatch(changeUserOrdersListIsLoad(false));
      })
      .catch((err) => {
        alert(err);
        dispatch(changeUserOrdersListIsLoad(false));
      });
  }
  return (
    <List
      dataArr={state.userOrders||[]}
      mainRows={["name", "time", "date", "town"]}
      getData={getUserOrdersFromServerToState}
      history={history}
      listIsLoad={state.userOrdersListIsLoad}
    />
  );
};

export default UserOrdersList;
