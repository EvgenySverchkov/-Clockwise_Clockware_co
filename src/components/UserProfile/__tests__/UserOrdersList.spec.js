import React from "react";
import {create, act} from "react-test-renderer";
import {Provider} from "react-redux";
import configMockStore from "redux-mock-store";

import UserOrdersList from "../UserOrdersList";

const mockStore = configMockStore();

it("Test of <UserOrdersList/>", async()=>{
    global.fetch = jest.fn(()=>Promise.resolve({json: jest.fn(()=>Promise.resolve())}));
    const store = mockStore({
        ordersReducer:{
            userOrders: [{
                id: 1,
                name: "Name",
                email: "admin@example.com",
                size: "middle",
                town: "Uzhorod",
                date: "2021-01-01",
                time: "13:00",
                masterId: 5,
                endTime: "15:00"
            }],
            userOrdersListIsLoad: false
        }
    });
    const mockPromise = {
        history: {}
    }
    let component
    await act(async ()=>{
        component = await create(
            <Provider store = {store}>
                <UserOrdersList {...mockPromise}/>
            </Provider>
        )
    })
    expect(component.toJSON()).toMatchSnapshot();
});