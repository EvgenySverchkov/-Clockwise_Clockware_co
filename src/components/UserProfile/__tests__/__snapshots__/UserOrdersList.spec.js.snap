// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <UserOrdersList/> 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table table-dark"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Time
        </th>
        <th>
          Date
        </th>
        <th>
          Town
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          style={
            Object {
              "wordWrap": "break-word",
            }
          }
        >
          Name
        </td>
        <td
          style={
            Object {
              "wordWrap": "break-word",
            }
          }
        >
          13:00
        </td>
        <td
          style={
            Object {
              "wordWrap": "break-word",
            }
          }
        >
          2021-01-01
        </td>
        <td
          style={
            Object {
              "wordWrap": "break-word",
            }
          }
        >
          Uzhorod
        </td>
        <td
          className="text-center"
        >
          <button
            className="btn btn-secondary"
            data-target="#fullInfoModal"
            data-toggle="modal"
            onClick={[Function]}
            type="button"
          >
            Show full info
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
